### ISSUE TITLE
[clang-tidy] Check request: readability-avoid-yoda-conditions

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#164770]
It would be good to have a check that converts [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) into normal conditions, as read into plain English from left to right.

Example:

```cpp
if (0 == foo())
{}
```

should be converted into:

```cpp
if (foo() == 0)
{}
```

In 2025, accidental assignments should no longer be a thing, tools have caught up. Compilers warn about it via `-Wparentheses`, active by default in Clang and via `-Wall` in GCC.

**Comment by vbvictor** [comment]
I guess we should not warn on cases like `if ((3.14 < y) && (y <= 42))`.
As first iteration we can only check if `if`-statement has single condition operator with a number on LHS and functionCall/variable on RHS.

Seems like an easy check, should be make it good first issue?

**Comment by vbvictor** [comment]
To work with clang-tidy, please read https://clang.llvm.org/extra/clang-tidy/Contributing.html and see how other checks are implemented (use clang-tidy tag on github)
